src/android/LocalyticsPlugin.java

```diff
@@ -906,20 +928,38 @@ public void run() {
             }
             return true;
         } else if (action.equals("triggerRegion")) {
-            if (args.length() == 2) {
+            if (args.length() >= 2) {
                 JSONObject region = args.getJSONObject(0);
                 String event = args.getString(1);
-                Localytics.triggerRegion(toCircularRegion(region), toEvent(event));
+                double latitude = args.optDouble(2);
+                double longitude = args.getDouble(3);
+                if (latitude != Double.NaN && longitude != Double.NaN) {
+                  Location location = new Location("");
+                  location.setLatitude(latitude);
+                  location.setLongitude(longitude);
+                  Localytics.triggerRegion(toCircularRegion(region), toEvent(event), location);
+                } else {
+                  Localytics.triggerRegion(toCircularRegion(region), toEvent(event), null);
+                }
                 callbackContext.success();
             } else {
                 callbackContext.error("Expected two arguments.");
             }
             return true;
         } else if (action.equals("triggerRegions")) {
-            if (args.length() == 2) {
+            if (args.length() >= 2) {
                 JSONArray regions = args.getJSONArray(0);
                 String event = args.getString(1);
-                Localytics.triggerRegions(toCircularRegions(regions), toEvent(event));
+                double latitude = args.optDouble(2);
+                double longitude = args.getDouble(3);
+                if (latitude != Double.NaN && longitude != Double.NaN) {
+                  Location location = new Location("");
+                  location.setLatitude(latitude);
+                  location.setLongitude(longitude);
+                  Localytics.triggerRegions(toCircularRegions(regions), toEvent(event), location);
+                } else {
+                  Localytics.triggerRegions(toCircularRegions(regions), toEvent(event), null);
```

Comment:
Same comment, log about no location

Reply:
Once I finish with updates I'll go through both sides and add a bunch of logging.  Looks like in general it is lacking